Api tuning

Gwellir opened the pull request:
добавляем полноценный CRUD к профилям пользователей и маршрутам
WIP (доработать правку маршрутов)

Gwellir committed:
add a way for a user to edit his profile data

Gwellir committed:
add multiple files upload for route creation via API

Gwellir committed:
fix import order in serializers.py

side2k commented on authapp/permissions.py:
```diff
@@ -0,0 +1,14 @@
+    """
+
+    def has_object_permission(self, request, view, obj):
+        if isinstance(request.user, AnonymousUser):
```
https://docs.djangoproject.com/en/3.1/ref/contrib/auth/#django.contrib.auth.models.User.is_authenticated

Gwellir replied:
благодарю, был уверен, что собственно делалось просто, но в рамках курса мы с оформленными пермишшенами не работали, поэтому не сложилось сразу

Gwellir committed:
fix authentication check in permissions.py
fix permission check never reaching has_object_permission for PUT or DELETE methods

Gwellir committed:
remove double permission check

Gwellir committed:
related permission fixes for authapp and ordersapp
